tests/test_D1_model_components.py

```diff
@@ -0,0 +1,147 @@
+import pytest
+import pandas as pd
+import os
+import json
+import oemof.solph as solph
+
+# internal imports
+import src.D1_model_components as D1
+from tests.constants import (
+    TEST_REPO_PATH
+)
+
+JSON_PATH = os.path.join(TEST_REPO_PATH, "inputs", "inputs_for_specific_tests",
+                         "mvs_config_for_D1_tests.json")
+
+
+# @pytest.fixture()
+# def get_json():
+#     with open(JSON_PATH) as json_file:
+#         dict_values = json.load(json_file)
+#     yield dict_values
+#
+#
+# @pytest.fixture()
+# def get_model():
+#     time_index = pd.date_range(start=pd.to_datetime("2018-01-01 00:00:00"),
+#                                end=pd.to_datetime("2018-12-31 23:00:00"),
+#                                freq="H")
+#     yield solph.EnergySystem(timeindex=time_index)
+
+
+class TestTransformerComponent:
```

Comment:
> TestTransformerComponent

Sounds good - you might want to add here the test checking if efficiencies can also be defined as a time series (heat pump), which can also be combined with multiple in/out busses

Reply:
Good point! I will add these tests.

Probably we need to add or rename the functions for doing so. In D1 there's only `transformer_constant_efficiency_fix()` and `transformer_constant_efficiency_optimize()`. However, I think @marc-juanpera  has already used the transformer with a time dependent efficiency - so maybe it's just the name of the function that's not correct.